lib/galaxy/model/__init__.py

```diff
@@ -1392,7 +1406,7 @@ class Job(Base, JobLike, UsesCreateAndUpdateTime, Dictifiable, Serializable):
     params: Mapped[Optional[str]] = mapped_column(TrimmedString(255), index=True)
     handler: Mapped[Optional[str]] = mapped_column(TrimmedString(255), index=True)
     preferred_object_store_id: Mapped[Optional[str]] = mapped_column(String(255))
-    object_store_id_overrides: Mapped[Optional[bytes]] = mapped_column(JSONType)
+    object_store_id_overrides: Mapped[Optional[STR_TO_STR_DICT]] = mapped_column(JSONType)
```

Comment:
```suggestion
    object_store_id_overrides: Mapped[Optional[STR_TO_STR_DICT]]
```
I think we don't need this: it's derived from the `type_annotation_map`.
Same for `transform` below.